.github/workflows/ci.yml

```diff
@@ -0,0 +1,445 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - staging
+      - feature*
+
+env:
+  # Caching
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  CACHIX_SIGNING_KEY: ${{ secrets.CACHIX_SIGNING_KEY }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NIX_CACHE_PRIV_KEY: ${{ secrets.NIX_CACHE_PRIV_KEY }}
+  NIX_CACHE_PUB_KEY: ${{ secrets.NIX_CACHE_PUB_KEY }}
+  NIX_CONFIG: access-tokens = github.com=${{ secrets.NIXPKGS_PRIVATE_PAT }}
+
+  # Other
+  GH_PROJECT_PATH: ${{ github.repository }}
+  GIT_SUBMODULE_STRATEGY: recursive
+  npm_config_cache: "${{ github.workspace }}/tmp/npm"
+  npm_config_prefer_offline: "true"
+
+jobs:
+  check-lint:
+    name: "Check / Lint"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: entrypoint
+
+      - name: Run linting
+        run: |
+          nix develop .#ci --command bash -c $'
+            npm run lint
+            npm run lint-shell
+          '
+
+  check-dry:
+    name: "Check / Dry Run"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: entrypoint
+
+      - name: Dry run
+        run: nix build .#docker --dry-run
+
+  build-pull:
+    name: "Build / Pull Request"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    needs:
+      - check-lint
+      - check-dry
+
+    if: github.ref == 'refs/heads/staging'
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: entrypoint
+
+      - name: Create pull request
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: |
+          git config --global --add safe.directory /__w/Polykey-Enterprise/Polykey-Enterprise
+
+          gh pr create \
+            --head staging \
+            --base master \
+            --title "ci: merge staging to master" \
+            --body "This is an automatic PR generated by the CI/CD pipeline. This will be automatically fast-forward merged if successful." \
+            --assignee "@me" \
+            --no-maintainer-edit || true
+
+          printf "Pipeline Attempt on $GITHUB_RUN_ID for $GITHUB_SHA\n\n$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" \
+          | gh pr comment staging \
+            --body-file - \
+            --repo "$GH_PROJECT_PATH"
+
+  build-dist:
+    name: "Build / Dist"
```

Comment:
Perhaps rename `Build / Dist` to something like `Build / Distribution` to make it more obvious of it's purpose.

Reply:
Eh its probably fine as is.